pkg/cluster/provider.go

```diff
@@ -0,0 +1,59 @@
+package cluster
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/watch"
+	"sigs.k8s.io/logical-cluster"
+)
+
+// Provider defines methods to retrieve, list, and watch fleet of clusters.
+// The provider is responsible for discovering and managing the lifecycle of each
+// cluster.
+//
+// Example: A Cluster API provider would be responsible for discovering and managing
+// clusters that are backed by Cluster API resources, which can live
+// in multiple namespaces in a single management cluster.
```

Comment:
nit: I guess technically we are not limited to a single mgmt cluster? (or are we?)

nit: let's add a bit of godoc to Get

Reply:
As long as the cluster names are "identifying" aka unique, then of course multiple mgmt clusters work too.

Have added go doc to Get.